test/test_distributions.py

```diff
@@ -4092,5 +4092,191 @@ def tearDown(self):
+                values = tuple(param[key] for key in keys)
+                if not all(isinstance(x, torch.Tensor) for x in values):
+                    continue
+                sample = Dist(**param).sample()
+                yield Dist, keys, values, sample
+
+    def test_sample(self):
+        for Dist, keys, values, sample in self._examples():
+
+            def f(*values):
+                param = dict(zip(keys, values))
+                dist = Dist(**param)
+                return dist.sample()
+
+            traced_f = torch.jit.trace(f, values, check_trace=False)
+
+            # FIXME Schema not found for node
+            xfail = [
+                Cauchy,  # aten::cauchy(Double(2,1), float, float, Generator)
+                Exponential,  # aten::exponential(Double(5, 5), float, Generator)
+                Geometric,  # aten::uniform(Double(3), float, float, Generator)
+                Gumbel,  # aten::uniform(Double(5, 5), float, float, Generator)
+                HalfCauchy,  # aten::cauchy(Double(2, 1), float, float, Generator)
+                Laplace,  # aten::uniform(Double(5, 5), float, float, Generator)
+                LowRankMultivariateNormal,  # aten::normal(Dynamic, float, float, Generator)
+                MultivariateNormal,  # aten::normal(Dynamic, float, float, Generator)
+                OneHotCategorical,  # aten::scatter(Double(2, 3), int, Dynamic, int)
+                Pareto,  # aten::exponential(Double(5, 5), float, Generator)
+                RelaxedBernoulli,  # aten::uniform(Double(3), float, float, Generator)
+                RelaxedOneHotCategorical,  # aten::uniform(Double(2, 3), float, float, Generator)
+                StudentT,  # aten::normal(Double(2, 3), float, float, Generator)
+                Uniform,  # aten::uniform(Double(5, 5), float, float, Generator)
+                Weibull,  # aten::exponential(Double(5, 5), float, Generator)
+            ]
+            if Dist in xfail:
+                print('        xfail reproducibility')
+                continue
+
+            with torch.random.fork_rng():
+                sample_0 = traced_f(*values)
+            sample_1 = traced_f(*values)
+            self.assertEqual(sample_0, sample_1)
+
+            # FIXME no nondeterministic nodes found in trace
+            xfail = [Beta, Dirichlet]
+            if Dist in xfail:
+                print('        xfail nondeterministic nodes')
+                continue
+            self.assertTrue(any(n.isNondeterministic() for n in traced_f.graph.nodes()))
+
+    def test_rsample(self):
+        for Dist, keys, values, sample in self._examples():
+            if not Dist.has_rsample:
+                continue
+
+            def f(*values):
+                param = dict(zip(keys, values))
+                dist = Dist(**param)
+                return dist.rsample()
+
+            traced_f = torch.jit.trace(f, values, check_trace=False)
+
+            # FIXME Schema not found for node
```

Comment:
Most of these failures appear to be due to our calling random generators with float args rather than tensor args. I think we can simply promote floats to tensors here, assuming all floats are constants.